packages/flutter/lib/src/widgets/animated_list.dart

```diff
@@ -228,15 +226,34 @@ class AnimatedListState extends State<AnimatedList> with TickerProviderStateMixi
   }
 
   _ActiveItem _removeActiveItemAt(List<_ActiveItem> items, int itemIndex) {
-    final int i = binarySearch(items, _ActiveItem.index(itemIndex));
+    final int i = _binarySearch(items, _ActiveItem.index(itemIndex));
     return i == -1 ? null : items.removeAt(i);
   }
 
   _ActiveItem _activeItemAt(List<_ActiveItem> items, int itemIndex) {
-    final int i = binarySearch(items, _ActiveItem.index(itemIndex));
+    final int i = _binarySearch(items, _ActiveItem.index(itemIndex));
     return i == -1 ? null : items[i];
   }
 
+  static int _binarySearch<T extends Comparable<T>>(List<T> sortedList, T value) {
```

Comment:
I wonder if this should live in foundation/collections.dart?

Reply:
I don't think we should add to the foundation API.